libstorage/api/types/types_localdevices.go

```diff
@@ -44,7 +44,9 @@ func (l *LocalDevices) MarshalText() ([]byte, error) {
 	keys := []string{}
 
 	for k := range l.DeviceMap {
-		keys = append(keys, k)
+		if k != "" {
```

Comment:
Hi @MatMaul,

I would swear on my life I removed the above addition recently. I remember it caused issues when removing the executor as the Marshal function removed the entries sans value. I decided it wasn't the place of a marshal function to do that. But I can find no record I actually made any changes. To be sure this file hasn't been modified in over a year!

The closest I can come is [this](https://github.com/codedellemc/libstorage/commit/2ef3d7edafe33c549fab9b8b016b5e0af66e27c0#diff-a1c38504f420f4720de9cf65512711ea):

![image](https://user-images.githubusercontent.com/101085/28987014-18fa4408-792f-11e7-853a-34c1c229df90.png)

That's in `drivers/storage/libstorage/libstorage_client_xcli.go`. It removes the devices without values in the map upon receiving the values back from the executor. Would this work for you?